lib/deployinator/controller.rb

```diff
@@ -147,7 +147,7 @@ def run(options, requires_lock = true)
       elsif options[:method].match(/force_builda/)
         env = "force asset rebuild"
       else
-        env = options[:method][/(dev|qa|production|princess|prod|webs|stage|staging|config)/i, 1] || "other"
+        env = options[:method][/(dev|qa|production|princess|preprod|prod|webs|stage|staging|config)/i, 1] || "other"
         env = "production" if env.match(/prod|webs/)
```

Comment:
@FacetGraph This is the problem with deployinator signaling a "Production Deploy Done" instead of "Preprod Deploy" so we need to take that into account.

Reply:
grrr yeah looking at that now it looks like you are right. But I tested this and saw PREPROD. Let me dive back into this and confirm whats going on.

Reply:
There is quite a few things that could cause problems here. You sort of have to name your environments with knowledge of this code in mind.
It would probably have been better to have a flag for production rather than try to do all this regex matching.
These lines are a little strange, like why would env ever be "force asset rebuild", seems kind of like overloading the env variable here

      if options[:method].match(/config_push/)                              
        env = options[:method].match(/prod/) ? "production" : "princess"    
      elsif options[:method].match(/force_builda/)                          
        env = "force asset rebuild"         if options[:method].match(/config_push/)                              
        env = options[:method].match(/prod/) ? "production" : "princess"    
      elsif options[:method].match(/force_builda/)                          
        env = "force asset rebuild"   

Anyways, I can fix this in the meantime by just adding a ^ to prod on the env.match line

Reply:
Yeah, there are a lot of etsy specific things in here.